war/src/main/java/io/lumeer/engine/controller/OrganizationFacade.java

```diff
@@ -157,16 +148,38 @@ public void dropOrganizationMetadata(final String organizationId, final String m
     *       id of the new organization to create
     * @param organizationName
     *       name of the new organization to create
+    * @throws OrganizationAlreadyExistsException
+    *       when organization with organizationId already exists
     */
-   public void createOrganization(final String organizationId, final String organizationName) {
+   public void createOrganization(final String organizationId, final String organizationName) throws OrganizationAlreadyExistsException {
+      if (dataStorage.collectionHasDocument(LumeerConst.Organization.COLLECTION_NAME, organizationIdFilter(organizationId))) {
+         throw new OrganizationAlreadyExistsException(ErrorMessageBuilder.organizationAlreadyExistsString(organizationId));
+      }
       DataDocument document = new DataDocument(LumeerConst.Organization.ATTR_ORG_ID, organizationId)
             .append(LumeerConst.Organization.ATTR_ORG_NAME, organizationName)
-            .append(LumeerConst.Organization.ATTR_ORG_DATA, new DataDocument())
-            .append(LumeerConst.Organization.ATTR_USERS, new DataDocument());
+            .append(LumeerConst.Organization.ATTR_ORG_DATA, new DataDocument());
       dataStorage.createDocument(LumeerConst.Organization.COLLECTION_NAME, document);
    }
 
    /**
+    * Changes organization id
+    *
+    * @param oldOrganizationId
+    *       id of the organization to change
+    * @param newOrganizationId
+    *       new id for organization
+    * @throws OrganizationAlreadyExistsException
+    *       when organization with newOrganizationId already exists
+    */
+   public void updateOrganizationId(final String oldOrganizationId, final String newOrganizationId) throws OrganizationAlreadyExistsException {
+      if (dataStorage.collectionHasDocument(LumeerConst.Organization.COLLECTION_NAME, organizationIdFilter(newOrganizationId))) {
+         throw new OrganizationAlreadyExistsException(ErrorMessageBuilder.organizationAlreadyExistsString(organizationId));
+      }
+      DataDocument document = new DataDocument(LumeerConst.Organization.COLLECTION_NAME, newOrganizationId);
+      dataStorage.updateDocument(LumeerConst.Organization.COLLECTION_NAME, document, organizationIdFilter(oldOrganizationId));
```

Comment:
Yes it is possible and it was done this way, but we have two options how to implement this. First is to have unique index on organization ID (same with project) and there's no need to check existency explicitly, but it cause duplicated index. Why? Because we have also compound index on organization ID + users so searching on organization ID can be covered by this index, but there's no mongo uniqueness checking and throwing exception. Second option is to remove index on organization ID and then we must explicitly check if it's already exists.
So what is better? Let's discuss it.. @livthomas